Kenny Gallardo
Build Your First Perceptron with Python
A Perceptron is a computer neuron that mimics human neurons. 

In [37]:
# Define the activation function (step function)
x1 = 1
x2 = 0

# Weights for the inputs
w1 = 0.8
w2 = -0.4

# Bias term
bias = -0.2

In [38]:
# Define the activation function (step function)
def step_activation(value):
    if value >= 0:
        return 1
    else:
        return 0

In [39]:
# Calculate the weighted sum (z)
z = (x1 * w1) + (x2 * w2) + bias
print("Weighted sum (z):", z)

output = step_activation(z)
print("Perceptron output:", output)

Weighted sum (z): 0.6000000000000001
Perceptron output: 1


Add a Markdown cell. Explain what the output means using complete sentences.

Your explanation must identify:

The weighed sum "Z" is 0.6000000000000001
Whether the weighted sum is greater than or equal to zero: the Weighted Sum is >= 0
Why the activation function returns 0 or 1: The activation return 1 because it's 0.6000000000000001 is >= 0.

In [40]:
def perceptron(x1, x2, w1, w2, bias):
    # weighted sum for the defined Perceptron (z)
    weighted_sum = (x1 * w1) + (x2 * w2) + bias
    step_output = step_activation(weighted_sum)
    return weighted_sum, step_output

In [41]:
weighted_sum, prediction = perceptron(1, 0, 0.8, -0.4, -0.2)
print("Weighted sum (z):", weighted_sum)
print("Perceptron output:", prediction)

Weighted sum (z): 0.6000000000000001
Perceptron output: 1


In [42]:
# Change Inputs x1 = 0 and x2 = 0
weighted_sum, prediction = perceptron(0, 0,0.8, -0.4, -0.2)
print("Weighted sum (z):", weighted_sum)
print("Perceptron output:", prediction)

Weighted sum (z): -0.2
Perceptron output: 0


In [43]:
# Change Inputs x1 = 0 and x2 = 1
weighted_sum, prediction = perceptron(0, 1, 0.8, -0.4, -0.2)
print("Weighted sum (z):", weighted_sum)
print("Perceptron output:", prediction)

Weighted sum (z): -0.6000000000000001
Perceptron output: 0


In [44]:
# Change Inputs x1 = 1 and x2 = 0
weighted_sum, prediction = perceptron(1, 0, 0.8, -0.4, -0.2)
print("Weighted sum (z):", weighted_sum)
print("Perceptron output:", prediction)

Weighted sum (z): 0.6000000000000001
Perceptron output: 1


In [45]:
# Change Inputs x1 = 1 and x2 = 1
weighted_sum, prediction = perceptron(1, 1, 0.8, -0.4, -0.2)
print("Weighted sum (z):", weighted_sum)
print("Perceptron output:", prediction)

Weighted sum (z): 0.2
Perceptron output: 1


In [46]:
# For Loop Randomized with imported random module

import random

def step_activation(z):
    return 1 if z >= 0 else 0

def perceptron(x1, x2, w1, w2, bias):
    # Weighted sum for the defined Perceptron (z)
    weighted_sum = (x1 * w1) + (x2 * w2) + bias
    step_output = step_activation(weighted_sum)
    return weighted_sum, step_output

# For loop to randomize the perceptron
for i in range(10):
    w1 = random.uniform(-1, 1)
    w2 = random.uniform(-1, 1)
    bias = random.uniform(-1, 1)

    weighted_sum, prediction = perceptron(1, 0, w1, w2, bias)

    print(f"Run {i + 1}:")
    print(f"  Weight 1 (w1): {w1:.2f}")
    print(f"  Weight 2 (w2): {w2:.2f}")
    print(f"  Bias: {bias:.2f}")
    print(f"  Weighted sum (z): {weighted_sum:.2f}")
    print(f"  Perceptron output: {prediction}")
    print("-" * 30)

Run 1:
  Weight 1 (w1): -0.26
  Weight 2 (w2): -0.05
  Bias: -0.17
  Weighted sum (z): -0.44
  Perceptron output: 0
------------------------------
Run 2:
  Weight 1 (w1): -0.03
  Weight 2 (w2): -0.94
  Bias: 0.33
  Weighted sum (z): 0.29
  Perceptron output: 1
------------------------------
Run 3:
  Weight 1 (w1): -0.09
  Weight 2 (w2): 0.35
  Bias: -0.26
  Weighted sum (z): -0.35
  Perceptron output: 0
------------------------------
Run 4:
  Weight 1 (w1): 0.09
  Weight 2 (w2): 0.63
  Bias: -0.07
  Weighted sum (z): 0.02
  Perceptron output: 1
------------------------------
Run 5:
  Weight 1 (w1): -0.55
  Weight 2 (w2): -0.62
  Bias: -0.15
  Weighted sum (z): -0.70
  Perceptron output: 0
------------------------------
Run 6:
  Weight 1 (w1): 0.97
  Weight 2 (w2): -0.32
  Bias: 0.92
  Weighted sum (z): 1.89
  Perceptron output: 1
------------------------------
Run 7:
  Weight 1 (w1): 0.40
  Weight 2 (w2): 0.37
  Bias: 0.00
  Weighted sum (z): 0.40
  Perceptron output: 1
---------------